cmd/cluster-init/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"unicode"
+
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	clusterName string
+	releaseRepo string
+}
+
+func (o options) String() string {
+	return fmt.Sprintf("%#v", o)
+}
+
+func parseOptions() (options, error) {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.clusterName, "cluster-name", "", "The name of the new cluster.")
+	fs.StringVar(&o.releaseRepo, "release-repo", "", "Path to the root of the openshift/release repository.")
+
+	return o, fs.Parse(os.Args[1:])
+}
+
+func validateOptions(o options) []error {
+	var errs []error
+	if o.clusterName == "" {
+		errs = append(errs, fmt.Errorf("--cluster-name must be provided"))
+	} else {
+		for _, char := range o.clusterName {
+			if unicode.IsSpace(char) {
+				errs = append(errs, fmt.Errorf("--cluster-name must not contain whitespace"))
+				break
+			}
+		}
+	}
+	if o.releaseRepo == "" {
+		errs = append(errs, fmt.Errorf("--release-repo must be provided"))
+	}
+	if o.clusterName != "" {
+		buildDir := buildFarmDirFor(o.releaseRepo, o.clusterName)
+		if _, err := os.Stat(buildDir); !os.IsNotExist(err) {
+			errs = append(errs, fmt.Errorf("build farm directory: %s already exists", o.clusterName))
+		}
+	}
+	return errs
+}
+
+func main() {
+	o, err := parseOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("cannot parse args: ", os.Args[1:])
+	}
+	validationErrors := validateOptions(o)
+	if len(validationErrors) > 0 {
+		logrus.Fatalf("validation errors: %v", validationErrors)
+	}
+
+	// Each step in the process is allowed to fail independently so that the diffs for the others can still be generated
+	errorCount := 0
+	for _, step := range []func(options) error{
+		initClusterBuildFarmDir,
+		updateSanitizeProwJobs,
+	} {
+		if err := step(o); err != nil {
+			logrus.WithError(err)
+			errorCount++
+		}
+	}
+	if errorCount > 0 {
+		logrus.Fatalf("Due to the %d error(s) encountered a PR will not be generated. The resulting files can be PR'd manually", errorCount)
+	}
+}
+
+func initClusterBuildFarmDir(o options) error {
+	buildDir := buildFarmDirFor(o.releaseRepo, o.clusterName)
+	logrus.Infof("Creating build dir: %s\n", buildDir)
```

Comment:
nit: you don't need the newline in these

Reply:
ah, didn't realize that, will update.